distributed/tests/test_batched.py

```diff
@@ -297,11 +297,5 @@ def raise_buffererror(*args, **kwargs):
             b.send("hello")
             b.send("world")
             await asyncio.sleep(0.020)
-            result = await comm.read()
```

Comment:
@TomAugspurger You wrote this test I touched here, so you may want to look at it: On BufferError, the first `comm.read` already fails.

Reply:
The test now catches another exception.

The original test was testing for https://github.com/dask/distributed/blob/eda9bcca8baa502481bf929f39340857da1ebf09/distributed/batched.py#L146

but with your implementation we're now hitting https://github.com/dask/distributed/blob/eda9bcca8baa502481bf929f39340857da1ebf09/distributed/comm/tcp.py#L184

Reply:
After looking into it a bit, I think this is a clash of philosophy and this test actually tests the opposite to what you are proposing here.
You suggest that e.g. a BufferError must not be retried but the implementation introduced in #4135 explicitly implements a retry mechansim for these failures.

What happens is: the first instance of the mock raises a `BufferError` which was supposed to be retried. Your implementation now, however, immediately closes the comm before anything can be retried.

If we allow for retries in other exception cases, the "solution" would be to change the mock to use a different exception class than BufferError. However, I think the intention was to explicitly deal with a buffer error :/

Reply:
> After looking into it a bit, I think this is a clash of philosophy and this test actually tests the opposite to what you are proposing here.

I think the code that re-tries here is deeply flawed, for one simply reason: one cannot safely re-try `write` if one does not know how much was written in the previous attempt. So I'd rather go ahead and delete all the retry code for the batched send. It is just plain wrong to retry here.

> You suggest that e.g. a BufferError must not be retried but the implementation introduced in #4135 explicitly implements a retry mechansim for these failures.

Yes, re-trying is just wrong.

Reply:
I added a second commit which removes the re-tries in `distributed.batched`.

I re-based on the latest master, so tests have a chance to run.